core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/CostBasedRegionPlanGenerator.scala

```diff
@@ -68,10 +69,16 @@ class CostBasedRegionPlanGenerator(
           })
           .filter(link => operatorIds.contains(link.fromOpId))
         val operators = operatorIds.map(operatorId => physicalPlan.getOperator(operatorId))
+        val materializedPorts: Set[GlobalPortIdentity] = matEdges.flatMap(link =>
+          List(
+            GlobalPortIdentity(link.fromOpId, link.fromPortId, input = false)
+          )
+        )
         Region(
           id = RegionIdentity(idx),
           physicalOps = operators,
-          physicalLinks = links
+          physicalLinks = links,
+          materializedPorts = materializedPorts
```

Comment:
Can we put materialization as a property of a port?

Reply:
Since we are currently using sink operators for materialization, and we have not finished associating storage on ports, I suggest we do it as part of the port storage project.

Reply:
I don't see a blocker or dependency here. It's a piece of information to be put somewhere. What's the benefit to keep it in a map of port ids, instead of keeping it on the actual object itself? 

For instance, compare the following design:

```
Class Student:
    Name
    ID
    Age 

Class Course:
    Students: List[Student]
    Has_job: List[StudentID]
```
Why not just put a has_job Boolean on Student object instead?

Also, a second thought, as the materialization is an execution-related information, we might should put it on the ExecutionConfig. And leave the Output port in the physical plan only describe the original physicalplan related information.

Reply:
After offline discussion, we agreed that materialzaition infomation belongs to Region instead of Physical Plan. We can keep it on the Region. 

Though please change it from `materializedPorts: List[GlobalPortId]` to `materializedPortIds: Set[GlobalPortId]`.

Reply:
Changed.